Source/Fluids/WarpXFluidContainer.cpp

```diff
@@ -196,9 +203,21 @@ void WarpXFluidContainer::InitData(int lev, amrex::Box init_box)
                 amrex::Real z = problo[0] + i * dx[0];
 #endif
 
+                // Lorentz transform z (from boosted to lab frame)
+                if (WarpX::gamma_boost > 1._rt){
+                    amrex::Real t_boost = gamma_boost*cur_time;
+                    z = gamma_boost*(z + beta_boost*clight*t_boost);
+                }
+
                 amrex::Real n = inj_rho->getDensity(x, y, z);
                 auto u = inj_mom->getBulkMomentum(x, y, z);
 
+                // Lorentz transform n, u (from lab to boosted frame)
+                if (WarpX::gamma_boost > 1._rt){
+                    n = gamma_boost*(n - beta_boost*n*u.z/clight);
+                    auto nuz = gamma_boost*(n*u.z - beta_boost*n*clight);
```

Comment:
This formula has a problem, because `n` on the right hand side is already the `n` in the boosted frame (as a consequence of line 217), whereas it should be the `n` in the lab frame. I would suggest doing things more explicitly:
```
n_lab = gamma_boost*(n - beta_boost*n*u.z/clight);
Real nuz_lab = gamma_boost*(n*u.z - beta_boost*n*clight);
u.z = nuz_lab/n_lab;
n = n_lab;

Reply:
Change included, fixed in the latest commit.